modules/gitops-argo-cd-dynamic-scaling.adoc

```diff
@@ -0,0 +1,112 @@
++
+.Example Argo CD YAML file with dynamic scaling enabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: openshift-gitops
+  namespace: openshift-gitops
+spec:
+  controller:
+    sharding:
+      dynamicScalingEnabled: true <1>
+      minShards: 1 <2>
+      maxShards: 3 <3>
+      clustersPerShard: 1 <4>
+----
+<1> Set `dynamicScalingEnabled` to `true` to enable dynamic scaling.
+<2> Set `minShards` to the minimum number of shards that you want to have. The value must be set to `1` or greater.
+<3> Set `maxShards` to the maximum number of shards that you want to have. The value must be greater than the value of `minShards`.
+<4> Set `clustersPerShard` to the number of clusters that you want to have per shard. The value must be set to `1` or greater.
+
+. Click *Save* to save the changes.
+
+* To enable dynamic scaling of shards using the command-line interface, complete the following steps:
+. Log in to the cluster using the `oc` command-line tool as a user with `cluster-admin` privileges.
+. Enable dynamic scaling by running the following command:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --type=merge --patch='{"spec":{"controller":{"sharding":{"dynamicScalingEnabled":true,"minShards":<value>,"maxShards":<value>,"clustersPerShard":<value>}}}}'
+----
++
+.Example command
+[source,terminal]
+----
+$ oc patch argocd openshift-gitops -n openshift-gitops --type=merge --patch='{"spec":{"controller":{"sharding":{"dynamicScalingEnabled":true,"minShards":1,"maxShards":3,"clustersPerShard":1}}}}' <1>
+----
++
+<1> The example command enables dynamic scaling for the `openshift-gitops` Argo CD instance in the `openshift-gitops` namespace, and sets the minimum number of shards to `1`, the maximum number of shards to `3`, and the number of clusters per shard to `1`.
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/openshift-gitops patched
+----
+
+. Verify that dynamic scaling is enabled by running the following command: 
++
+[source,terminal]
+----
+$ oc get argocd <argocd_instance> -n <namespace> -o jsonpath='{.spec.controller.sharding}'
+----
++
+.Example command 
+[source,terminal]
+----
+$ oc get argocd openshift-gitops -n openshift-gitops -o jsonpath='{.spec.controller.sharding}'
+----
++
+.Example output when dynamic scaling of shards is enabled
+[source,terminal]
+----
+{"dynamicScalingEnabled":true,"minShards":1,"maxShards":3,"clustersPerShard":1}
+----
+
+.. Alternatively, you can verify that dynamic scaling is enabled by checking the configured `spec.controller.sharding` properties in the configuration `YAML` file of the Argo CD instance on the {OCP} web console.
+
+[NOTE]
+==== 
+Once you enable dynamic scaling, you cannot manually modify the shard count. The system automatically adjusts the number of shards based on the number of clusters managed by the Argo CD Application Controller at a given time.
+====
```

Comment:
Move this note under the introduction so that it is outside the new two sections.

Reply:
ACK.

Addressed.